### ISSUE TITLE
could not add any kind of entry in VGUI

### ISSUE THREAD

**Issue by fa0123** [ValveSoftware/halflife#1841]
hi i got a problem while im adding some new entries "PasswordEntry" to be usable for password "" command like "NameEntry" in VGUI Build Mode Editor
looks like theres no way to add new options
my version is :

Protocol version 47
Exe version 1.1.2.5 (cstrike)
Exe build: 10:56:59 Oct  3 2005 (3266)

is there anyway to i customize my VGUI Options menu in cs1.6?

tnx in advance.

**Comment by tschumann** [comment]
Are you using the Ctrl+Alt+Shift+B or whatever it is that works in Source engine games? It's probably a very outdated version of the VGUI Build Mode Editor so I wouldn't be surprised if editing doesn't work.
Can you manually modify the .res files in the resource/ directory to do what you want?

**Comment by fa0123** [comment]
yes i used VGUI Build Mode Editor with this shortcut key Ctrl+Alt+Shift+B

i can modify in .res but my custom entry will not work
example of mine:

	"PasswordEntry"
	{
		"ControlName"		"CCvarTextEntry"
		"fieldName"		"PasswordEntry"
		"xpos"		"21"
		"ypos"		"64"
		"wide"		"225"
		"tall"		"24"
		"autoResize"		"0"
		"pinCorner"		"0"
		"visible"		"1"
		"enabled"		"1"
		"tabPosition"		"1"
		"paintbackground"		"1"
		"textHidden"		"0"
		"editable"		"1"
		"maxchars"		"31"
		"NumericInputOnly"		"0"
		"unicode"		"1"
	}

its only work with this command:

	"NameEntry"
	{
		"ControlName"		"CCvarTextEntry"
		"fieldName"		"NameEntry"

NameEntry = name "" (in console)

i want to make something like this:
PasswordEntry = password "" (in console)

**Comment by tschumann** [comment]
Yeah the version of VGUI that GoldSource uses was probably last updated in 2003 or 2004 so it probably doesn't support PasswordEntry.

**Comment by fa0123** [comment]
so which version support
i think theres no way ...

**Comment by tschumann** [comment]
No idea but yeah, without the source code there's probably no way.

**Comment by 0nurky** [comment]
Are you sure? Wasn't there a option in Counter-Strike to add a password? Isn't that a PasswordEntry acting thing in VGUI?

**Comment by fa0123** [comment]
actually not, may i confused about it

**Comment by 0nurky** [comment]
I remember there was at the end of options, you set a password so that your younger family members can't play the game.

**Comment by SamVanheer** [comment]
You can add new elements but they won't do anything without code to handle it. You can't do that since the code would need to be in GameUI.dll, which is closed source.

**Comment by 0nurky** [comment]
That's bad to hear. It's weird from the side of Valve, why keep the source code of an old engine, I guess they use some of that code in the source engine though...

**Comment by SamVanheer** [comment]
A good chunk of what's in GameUI was released with the Alien Swarm SDK, but not everything is in that release, notably the Condition Zero specific UI.

**Comment by 0nurky** [comment]
Would be great if someone, somewhere, ported Panorama UI to the old GoldSrc games.
XD
